tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -13,44 +13,122 @@ else
   $script:BootStrapRepo = $existingRepos[0].Name
 }
 
+# Check if current user is Admin to decide on cache path
+$script:IsAdmin = $false
+if ($IsWindows)
+{
+  If (([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
+  {
+    $script:IsAdmin = $true
+  }
+}
+else {
+  # on Linux, tests run via sudo will generally report "root" for whoami
+  if ( (whoami) -match "root" ) 
+  {
+    $script:IsAdmin = $true
+  }
+}
+
+# Get profile cache path
 function Get-ProfileCachePath
 {
-  $ProfileCache = Join-Path -path $env:LOCALAPPDATA -childpath "Microsoft\AzurePowerShell\ProfileCache"
+  if ($IsWindows)
+  {
+    $ProfileCache = Join-Path -path $env:LOCALAPPDATA -childpath "Microsoft\AzurePowerShell\ProfileCache"
+    if ($script:IsAdmin)
+    {
+      $ProfileCache = Join-Path -path $env:ProgramData -ChildPath "Microsoft\AzurePowerShell\ProfileCache"
+    }
+  }
+  else
+  {
+    $ProfileCache = "$HOME/.config/Microsoft/AzurePowerShell/ProfileCache" 
+  }
   return $ProfileCache
 }
 
-# Make Rest-Call
-function Get-RestResponse
+# Function to find the latest profile map from cache
+function Get-LatestProfileMapPath
 {
-  $response = Invoke-RestMethod -ea SilentlyContinue -Uri $PSProfileMapEndpoint -ErrorVariable RestError -OutFile "$ProfileCache\ProfileMap.Json"
-  return $RestError
+  $ProfileCache = Get-ProfileCachePath
+  if (-Not (Test-Path $ProfileCache))
+  {
+    return
+  }
+
+  $ProfileMapPaths = Get-ChildItem $ProfileCache
+
+  if ($ProfileMapPaths -eq $null)
+  {
+    return
+  }
+
+  $LatestMapPath = $ProfileMapPaths[0]
+  foreach ($ProfileMapPath in $ProfileMapPaths)
```

Comment:
We are using incrementing ordinal numbers to name profilemaps postfixed with Etags. This helps us to retrieve the latest map using the largest number in cache.